### PR TITLE
feat: ordering of only relevant loops(no visual)

### PR DESCRIPTION
**Pull request by moyne**
Additional parameter that can be received on a layer that defines a set of loop to be left unordered
Blocking loop factorization and ordering of such loops
(no update to the visual bar yet)

### CODE REVIEW

File: zigzag/classes/opt/temporal/loma/multipermute.py [hunk 1]
```diff
@@ -59,19 +59,21 @@ def init(multiset):
...
-def visit(h):
+def visit(h,original_multiset,unordered_loops):
     o = h
     l = []
     while o is not None:
         l.append(o.value)
         o = o.next
-    return l
+    # place unordered loops at the end
+    return [(name,size) for name,size in original_multiset if name in unordered_loops]+l
```
Comment (asyms) [inline zigzag/classes/opt/temporal/loma/multipermute.py:69]: Does this not place the "unordered loops" at the start (i.e. innermost loops) of the permutation? As opposed to "at the end" as you state in the PR?

**Review by asyms — COMMENTED** [review]
One small question regarding the unordered loops. Are they placed as the start (innermost) or end (outermost) loop?

**Comment by moyne** [comment]
Hi, thank you for the question, they are placed as the innermost ones (so at the end of the computation, where the tensors should be allocated already at innermost memory level)

**Review by asyms — CHANGES_REQUESTED** [review]
I'm not entirely sure how you would use this. Can you add some test scenarios that use it in a way in a new folder here: https://github.com/KULeuven-MICAS/zigzag/tree/master/tests/main ?

**Comment by asyms** [comment]
@moyne Could you please resolve the conflicts? I still believe your additions would add significant value to our project!

**Comment by moyne** [comment]
Hi @asyms sorry for the late reply, I'll try to fix it maybe even with a new PR that considers the case for output stationary workloads separetely, what do you think about it? Maybe it would be cool to define the type of workload supported and put constraints on the memory allocator/loop factorization knowing these info

**Comment by asyms** [comment]
Hi @moyne,

There has been more work related to this since this PR was opened. See: https://github.com/KULeuven-MICAS/zigzag/pull/75

It would indeed be great if you can take a look at that PR and the state of the codebase and submit another PR in case there is anything missing functionality-wise.

P.S.: Marian also mentioned you have some systolic array modelling changes, please let us know how we can help to upstream these!

**Pull request closed without merging**